packages/dart/lib/src/network/parse_dio_client.dart

```diff
@@ -31,7 +31,9 @@ class ParseDioClient extends ParseClient {
           data: dioResponse.data!, statusCode: dioResponse.statusCode!);
     } on dio.DioError catch (error) {
       return ParseNetworkResponse(
-          data: error.response?.data, statusCode: error.response!.statusCode!);
+        data: error.response?.data ?? _fallbackErrorData,
+        statusCode: error.response!.statusCode!,
```

Comment:
I ran this code without internet connection and got null error in statusCode!
I suggest you change the code like this

```
return ParseNetworkResponse(
  data: error.response?.data ?? _fallbackErrorData,
  statusCode: error.response?.statusCode! ?? -1,
);
```